main.go

```diff
@@ -373,6 +323,72 @@ func mainErr(args []string) error {
 	return nil
 }
 
+// toolexecCmd builds an *exec.Cmd which is set up for running "go <command>"
+// with -toolexec=garble and the supplied arguments.
+//
+// Note that it uses and modifies global state; in general, it should only be
+// called once from mainErr in the top-level garble process.
+func toolexecCmd(command string, args []string) (*exec.Cmd, error) {
+	if !goVersionOK() {
+		return nil, errJustExit
+	}
+	// Split the flags from the package arguments, since we'll need
+	// to run 'go list' on the same set of packages.
+	flags, args := splitFlagsFromArgs(args)
+	for _, f := range flags {
+		switch f {
+		case "-h", "-help", "--help":
+			return nil, flag.ErrHelp
+		}
+	}
+
+	if err := setOptions(); err != nil {
+		return nil, err
+	}
+
+	// Note that we also need to pass build flags to 'go list', such
+	// as -tags.
+	cache.BuildFlags = filterBuildFlags(flags)
+	if command == "test" {
+		cache.BuildFlags = append(cache.BuildFlags, "-test")
+	}
+
+	if err := setGoPrivate(); err != nil {
+		return nil, err
+	}
+
+	var err error
+	cache.ExecPath, err = os.Executable()
+	if err != nil {
+		return nil, err
+	}
+
+	if err := setListedPackages(args); err != nil {
+		return nil, err
+	}
+
+	if sharedTempDir, err = saveShared(); err != nil {
```

Comment:
I think this should be done like in line 361, mutating global state inside a if condition feels a little weird.

Reply:
Sure. This is just moved code, though. I really wish github showed that better.